packages/api/src/features/receive/handler.ts

```diff
@@ -123,100 +73,36 @@ async function asyncReceiveNid(payloadStr: string, openCRVSToken: string) {
   })
 }
 
-function decryptData(requestData: Buffer): string {
-  const keyDemiliterIndex: number = requestData.indexOf(KEY_SPLITTER)
-  if (keyDemiliterIndex < 0) {
-    throw new Error('Imporper encrypted data format')
-  }
-
-  let encryptedSymmetricKey: Buffer
-  let nonce: Buffer
-  let aad: Buffer = Buffer.alloc(0)
-  let encryptedData: Buffer
-  let authTag: Buffer
-
-  if (requestData.indexOf(VERSION_RSA_2048) === 0) {
-    encryptedSymmetricKey = requestData.subarray(
-      THUMBPRINT_LENGTH + VERSION_RSA_2048.length,
-      keyDemiliterIndex
-    )
-    nonce = requestData.subarray(
-      keyDemiliterIndex + KEY_SPLITTER.length,
-      keyDemiliterIndex + KEY_SPLITTER.length + NONCE_SIZE
-    )
-    aad = requestData.subarray(
-      keyDemiliterIndex + KEY_SPLITTER.length,
-      keyDemiliterIndex + KEY_SPLITTER.length + AAD_SIZE
-    )
-    encryptedData = requestData.subarray(
-      keyDemiliterIndex + KEY_SPLITTER.length + AAD_SIZE,
-      requestData.length - GCM_TAG_LENGTH
-    )
-    authTag = requestData.subarray(
-      requestData.length - GCM_TAG_LENGTH,
-      requestData.length
-    )
-
-    logger.info(`encKey Length: ${encryptedSymmetricKey.length}`)
-  } else if (IS_THUMBRPINT) {
-    encryptedSymmetricKey = requestData.subarray(
-      THUMBPRINT_LENGTH,
-      keyDemiliterIndex
-    )
-    encryptedData = requestData.subarray(
-      keyDemiliterIndex + KEY_SPLITTER.length + AAD_SIZE,
-      requestData.length - GCM_TAG_LENGTH
-    )
-    authTag = requestData.subarray(
-      requestData.length - GCM_TAG_LENGTH,
-      requestData.length
-    )
-    nonce = encryptedData.subarray(
-      encryptedData.length - GCM_TAG_LENGTH,
-      encryptedData.length
-    )
-  } else {
-    encryptedSymmetricKey = requestData.subarray(0, keyDemiliterIndex)
-    encryptedData = requestData.subarray(
-      keyDemiliterIndex + KEY_SPLITTER.length,
-      requestData.length - GCM_TAG_LENGTH
-    )
-    authTag = requestData.subarray(
-      requestData.length - GCM_TAG_LENGTH,
-      requestData.length
-    )
-    nonce = encryptedData.subarray(
-      encryptedData.length - GCM_TAG_LENGTH,
-      encryptedData.length
-    )
-  }
-  const opencrvsPrivKey: forge.pki.rsa.PrivateKey = forge.pki.privateKeyFromPem(
-    OPENCRVS_PRIV_KEY
-  )
-  const decryptedSymmetricKey = opencrvsPrivKey.decrypt(
-    encryptedSymmetricKey.toString('binary'),
-    ASYMMETRIC_ALGORITHM,
-    {
-      md: forge.md.sha256.create(),
-      mgf1: {
-        md: forge.md.sha256.create()
+async function putUINTokenToHIMMediator(
+  openCRVSAuthToken: string,
+  uinToken: string,
+  birthRegNo: string
+) {
+  try {
+    const nationalIdOpenHIMMediatorResponse = await fetch(
+      OPENHIM_MEDIATOR_URL,
+      {
+        method: 'POST',
+        body: JSON.stringify({
+          BRN: birthRegNo,
+          UINTOKEN: uinToken
```

Comment:
@lalithkota could you send the RID here please?

Reply:
@lalithkota so I imagine the item will be named MOSIP_AID and come independently in the previous fetch as per our conversation earlier today